source/our_tools/tractoflow.rst

```diff
@@ -3,4 +3,220 @@
+
+            # Wait for the node to be allocated to you.
+            # If our lab lacks priority and it is too long, you can try with -c 16 --mem 16Gb.
+            # When allocation is granted, you should see:
+            #   salloc: Pending job allocation xxx
+            #   salloc: job xxx queued and waiting for resources
+            #   salloc: job xxx has been allocated resources
+            #   salloc: Granted job allocation xxx
+            #   salloc: Waiting for resource configuration
+            #   salloc: Nodes yyym are ready for job
+
+            # Then type the tractoflow command with --help at the end. Tractoflow's help should be printed.
+
+            # Then, you may type the real command. Make sure it starts running. Once sure that it found the data, the img, the code, you can kill it by pressing ctrl-c.
+
+       If it fails:
+
+         - Is the data binding correctly (see point 5)?
+         - If one process fails, you should see a warning such as ``[11/53e26e] NOTE: Process `Bet_Prelim_DWI (101309)` terminated with an error exit status (127) -- Execution is retried (3)``. You can then check in the associated folder's log to see the error. For example:
+
+         .. code-block:: bash
+
+            ls -a work/11/53e26e*/  # Check that files are there
+            cat work/11/53e26e*/.command.log  # Check the error
+
+    #. Last decision to take on the cluster: Do you need 1 node or multiple nodes? Beyond 20 subjects or so, we recommend using multiple nodes. See the HPC part of tractoflow's `launch page <https://tractoflow-documentation.readthedocs.io/en/latest/pipeline/launch.html>`_. Depending on the cluster you are using (beluga, cedar, etc.), you have access to different types of nodes. Keeping the option "mem=0" in your sbatch (see next step) means you accept any node. If your data is very heavy (ex, HCP data), you might need to restric to the nodes with more RAM. See the `memory section here <https://docs.computecanada.ca/wiki/Running_jobs#Memory>`_ for more information.
+
+    #. We have all the ingredients to prepare the final sbatch. Here, we ask for 4 nodes, with 32 threads each and 116Gb of RAM each (see the -with-mpi option). Create a file cmd_my_tractoflow.sh with the following.
+
+        .. code-block:: bash
+
+            #!/bin/sh
+            #SBATCH --mail-user=YOUR_EMAIL
+            #SBATCH --mail-type=BEGIN
+            #SBATCH --mail-type=END
+            #SBATCH --mail-type=FAIL
+            #SBATCH --mail-type=REQUEUE
+            #SBATCH --mail-type=ALL
+
+            #SBATCH --nodes=2
+            #SBATCH --cpus-per-task=32
+            #SBATCH --mem=0
+            #SBATCH --time=24:00:00
+
+            export NXF_CLUSTER_SEED=$(shuf -i 0-16777216 -n 1)
+
+            srun (copy your nextflow command from point 9) -with-mpi
+
+    #. Finally launch your sbatch! Yeah!
+
+        .. code-block:: bash
+
+            sbatch -A def-descotea cmd_my_tractoflow.sh
```

Comment:
or rrg-descotea if you want to use the allocation lon beluga.

Reply:
I ran it on beluga and I used def-descotea and it worked... ?

Reply:
Yes but we have the allocation now. The name of this allocation is rrg-descotea